### PR TITLE
continue in pow loop instead of returning

### LINKED ISSUE #9537: Why does the POW authorship task prematurely terminate?
https://github.com/paritytech/substrate/blob/777034d4d799fbb727d731d5b157e8a4ebaaf1d8/client/consensus/pow/src/lib.rs#L602-L604

If for some reason the metadata hasn't been updated with a new a block, the authorship task is prematurely terminated, without any error messages or documentation for this behaviour

cc @sorpaas

### PR DESCRIPTION
**Pull request by OAyomide**
It replaces the return statement with a continue, to prevent the node from terminating early. Fixes #9537

### CODE REVIEW

**Review by seunlanlege — CHANGES_REQUESTED** [review]
There are other return statements in the task, replace them with `continue`

File: client/consensus/pow/src/lib.rs [hunk 1]
```diff
@@ -560,7 +560,8 @@ where
 			if sync_oracle.is_major_syncing() {
 				debug!(target: "pow", "Skipping proposal due to sync.");
 				worker.lock().on_major_syncing();
-				return
+				debug!(target: "pow", "SyncOracle is syncing.");
```
Comment (seunlanlege) [inline client/consensus/pow/src/lib.rs:563]: no need for this log line
```suggestion
```

File: client/consensus/pow/src/lib.rs [hunk 2]
```diff
@@ -572,7 +573,8 @@ where
 						 Select best chain error: {:?}",
 						err
 					);
-					return
+					warn!(target: "pow", "Unable to seal block");
```
Comment (seunlanlege) [inline client/consensus/pow/src/lib.rs:576]: also no need for this log line
```suggestion
```

File: client/consensus/pow/src/lib.rs [hunk 3]
```diff
@@ -584,11 +586,14 @@ where
 					 Probably a node update is required!",
 					err,
 				);
-				return
+				warn!(target: "pow", "Still waiting for seal for block {}", best_hash);
```
Comment (seunlanlege) [inline client/consensus/pow/src/lib.rs:589]: this as well
```suggestion
```

File: client/consensus/pow/src/lib.rs [hunk 4]
```diff
@@ -584,11 +586,14 @@ where
...
+				continue
 			}
 
 			if worker.lock().best_hash() == Some(best_hash) {
-				return
+				debug!(target: "pow", "Got the best hash");
```
Comment (seunlanlege) [inline client/consensus/pow/src/lib.rs:594]: double logs?
```suggestion
```

File: client/consensus/pow/src/lib.rs [hunk 5]
```diff
@@ -603,7 +608,8 @@ where
 						 Fetch difficulty failed: {:?}",
 						err,
 					);
-					return
+					warn!(target: "pow", "Still waiting for seal for block {}", best_hash);
```
Comment (seunlanlege) [inline client/consensus/pow/src/lib.rs:611]:
```suggestion
```

File: client/consensus/pow/src/lib.rs [hunk 6]
```diff
@@ -619,7 +625,8 @@ where
 						 Creating inherent data providers failed: {:?}",
 						err,
 					);
-					return
+					warn!(target: "pow", "Still waiting for seal for block {}", best_hash);
```
Comment (seunlanlege) [inline client/consensus/pow/src/lib.rs:628]:
```suggestion
```

File: client/consensus/pow/src/lib.rs [hunk 7]
```diff
@@ -632,7 +639,8 @@ where
 						 Creating inherent data failed: {:?}",
 						e,
 					);
-					return
+					warn!(target: "pow", "Still waiting for seal for block {}", best_hash);
```
Comment (seunlanlege) [inline client/consensus/pow/src/lib.rs:642]:
```suggestion
```

File: client/consensus/pow/src/lib.rs [hunk 8]
```diff
@@ -652,7 +660,8 @@ where
 						 Creating proposer failed: {:?}",
 						err,
 					);
-					return
+					warn!(target: "pow", "Still waiting for seal for block {}", best_hash);
```
Comment (seunlanlege) [inline client/consensus/pow/src/lib.rs:663]:
```suggestion
```

File: client/consensus/pow/src/lib.rs [hunk 9]
```diff
@@ -668,7 +677,8 @@ where
 						 Creating proposal failed: {:?}",
 						err,
 					);
-					return
+					warn!(target: "pow", "Still waiting for seal for block {}", best_hash);
```
Comment (seunlanlege) [inline client/consensus/pow/src/lib.rs:680]:
```suggestion
```

**Review by seunlanlege — CHANGES_REQUESTED** [review]

**Review by seunlanlege — APPROVED** [review]
Perfect

**Comment by seunlanlege** [comment]
pinging @sorpaas @bkchr @gilescope

**Review by bkchr — APPROVED** [review]
Please format properly

**Comment by bkchr** [comment]
CI fmt check is still not happy. You need to use `nightly` rustfmt.

**Review by sorpaas — APPROVED** [review]

**Review by sorpaas — APPROVED** [review]

**Comment by sorpaas** [comment]
bot merge

**Pull request merged**